app/controllers/users_controller.rb

```diff
@@ -1,52 +1,77 @@
 # frozen_string_literal: true
 
 class UsersController < ApplicationController
-  before_action :set_user, only: [:destroy]
-  before_action :require_admin
+  before_action :set_user, only: [:destroy, :edit, :update]
 
   # GET /users
   def index
+    authorize User
```

Comment:
For all the authorize checks - if the user isn't authorize then we'll get an application exception at the moment I think (I'll try  it out shortly to check) - I  guess we might just want to redirect to the home page with an unauthorized flash message.

Reply:
Yep good point, will change that

Reply:
This now redirects to root on any pundit error, see application_controller